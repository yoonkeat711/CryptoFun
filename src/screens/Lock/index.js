import React, {useState, useEffect} from 'react';
import {View, TextInput, Text, TouchableOpacity, Alert} from 'react-native';
import {Base64} from 'js-base64';

const Lock = ({navigation}) => {
  const [password, setPassword] = useState(null);
  const hardcodePassword = 'abc123';

  const onPressEnterPassword = async () => {
    const encryptPassword = Base64.encode(password);
    const hardcodeEncryptedPassword = Base64.encode(hardcodePassword);
    if (encryptPassword === hardcodeEncryptedPassword) {
      navigation.goBack();
    } else {
      Alert.alert('Error', 'Password Incorrect!');
    }
  }

  return (
    <View style={{flex: 1, backgroundColor: 'white', justifyContent: 'center'}}>
      <Text style={{alignSelf: 'center', paddingBottom: 8}}>{"Password"}</Text>
      <TextInput value={password}
                 onChangeText={value => setPassword(value)}
                 textAlign={'center'}
                 style={{alignSelf: 'center', borderWidth: 1, borderColor: 'grey', width: '50%', height: 30}}
      />
      <TouchableOpacity style={{alignSelf: 'center', width: '50%', paddingTop: 8}} onPress={onPressEnterPassword}>
        <Text style={{alignSelf: 'center'}}>{"Enter"}</Text>
      </TouchableOpacity>
    </View>
  );
}
export default Lock;

