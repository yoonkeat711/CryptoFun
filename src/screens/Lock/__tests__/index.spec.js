import renderer from 'react-test-renderer';
import Lock from '../index';
import React from 'react';

test('renders correctly', () => {
  const tree = renderer.create(<Lock />).toJSON();
  expect(tree).toMatchSnapshot();
})
