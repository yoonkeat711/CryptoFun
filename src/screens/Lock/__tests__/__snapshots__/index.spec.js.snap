// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "paddingBottom": 8,
      }
    }
  >
    Password
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "grey",
        "borderWidth": 1,
        "height": 30,
        "width": "50%",
      }
    }
    textAlign="center"
    underlineColorAndroid="transparent"
    value={null}
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "opacity": 1,
        "paddingTop": 8,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      Enter
    </Text>
  </View>
</View>
`;
